### PR TITLE
[LICM] Fix an ASAN use-after-free bug in rewrite

### PR DESCRIPTION
**Pull request by shajrawi**
see https://ci.swift.org/job/oss-swift-incremental-ASAN-RA-osx/1968

### CODE REVIEW

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

File: lib/SILOptimizer/LoopTransforms/LICM.cpp [hunk 1]
```diff
@@ -216,14 +227,20 @@ static bool sinkInstruction(DominanceInfo *DT,
 
   bool Changed = false;
   for (auto *ExitingBB : ExitingBBs) {
+    SmallVector<SILBasicBlock *, 8> BBSuccessors;
```
Comment (gottesmm) [inline lib/SILOptimizer/LoopTransforms/LICM.cpp]: Can you hoist this SmallVector our of the loop?

Comment (gottesmm) [reply lib/SILOptimizer/LoopTransforms/LICM.cpp]: To be clear, I am talking about hoisting it out of the loop and then clearing before each loop iteration.

Comment (shajrawi) [reply lib/SILOptimizer/LoopTransforms/LICM.cpp]: One exiting bb covers most loops, in fact we didn't bother support multiple basic blocks before the refactoring, loop will execute 1-2 iterations in the vast majority of cases, thought this is shorter / more readable.

**Pull request merged**